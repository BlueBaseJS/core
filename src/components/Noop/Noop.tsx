import React from 'react';

export interface NoopProps {
	children?: React.ReactNode,
}

/**
 * # 😐 Noop
 *
 * A component that does... nothing!
 */
export const Noop: React.FunctionComponent<NoopProps> = ({ children }) => {

	if (children && React.isValidElement(children)) {
		return children;
	}

	return null;
};
