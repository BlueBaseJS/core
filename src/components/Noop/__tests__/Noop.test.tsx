import { Noop } from '../Noop';
import React from 'react';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
	const component = renderer.create(
		<Noop />,
	);
	const tree = component.toTree();
	expect(tree).toMatchSnapshot();
});