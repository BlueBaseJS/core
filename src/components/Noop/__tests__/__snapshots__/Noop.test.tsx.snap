// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {},
  "rendered": null,
  "type": [Function],
}
`;
